code/data_setup/parse_alov_bb.py

```diff
@@ -0,0 +1,90 @@
+"""A script for parsing the Alov bounding box `.ann` files."""
+
+import sys
+import os
+import itertools
+import shutil
+import pandas as pd
+
+def parse_file(bbox_dir, filename): 
+    """Parse an individual `.ann` file and output the relevant elements.
+
+    Args: 
+    ----
+        bbox_dir: str
+            Holds the directory path to the `.ann` file. 
+        filename: str
+            Holds the name of the `.ann` file. 
+
+    Output: 
+    ------
+        output_lst: list of lists 
+            Holds the elements grabbed from the `.ann` file. 
+    """
+
+    parsed_df = pd.read_table(bbox_dir + filename, sep=' ', header=None)
+    parsed_df['filename'] = filename
+    parsed_df['directory'] = parsed_df['filename'].apply(lambda fn: fn.split('_')[0])
+
+    return parsed_df
+
+def cp_files(filepaths_df, input_dir, output_dir): 
+    """Copy over the files given by the attributes in the dataframe. 
+
+    Args: 
+    -----
+        filepaths_df: pandas DataFrame
+            Holds 3 columns, each of which holds a piece that 
+            determines the input filepaths. We'll copy the files 
+            at these filepaths to the output_dir.
+        input_dir: str
+            Holds the first part of the input directory path, used 
+            for obtaining the full path pointing to the files to copy. 
+        output_dir: str
+            Holds the directory of where to copy the file. 
+    """
+    
+    # The video numbers are zero padded on the left. 
+    filepaths_df['vid_num'] = filepaths_df['frame'].astype(str).apply(lambda 
+            frame: frame.zfill(8))
+
+    # Need to replace the filename ext. in the DataFrame with `.jpg` and 
+    # provide the full relative path point to what to move. 
+    filepaths_df['filename'] = filepaths_df['filename'].apply(lambda filename: 
+            filename.replace('.ann', ''))
+    filepaths_df['input_filepath'] = input_dir + \
```

Comment:
Turns out that `.assign` on a `DataFrame` still tries to perform array like operations (e.g. not row by row, but adding two arrays). So, throwing `join` around it simply gives the error that `join` doesn't take a `Series` object, but expects a string. I tried a couple of other things, none of which worked (`eval` and `df.apply`), so I'm going to call it a day. If it's important, happy to come back and spend more time figuring out a way to (hopefully) make it look more clean.